numpy/ma/extras.py

```diff
@@ -626,10 +626,14 @@ def median(a, axis=None, out=None, overwrite_input=False, keepdims=False):
            fill_value = 1e+20)
 
     """
-    if not hasattr(a, 'mask') or np.count_nonzero(a.mask) == 0:
-        return masked_array(np.median(getdata(a, subok=True), axis=axis,
-                            out=out, overwrite_input=overwrite_input,
-                            keepdims=keepdims), copy=False)
+    if not hasattr(a, 'mask'):
```

Comment:
why was the count_nonzero removed?
its performance impact is practically zero but the gain on empty mask is huge